### PR TITLE
Remove more JIT LEGACY_BACKEND tendrils

### PR DESCRIPTION
**Pull request by BruceForstall**
1. Remove armlb testing from netci.groovy
2. Remove legacy jit exclusions from tests.lst
3. Remove some unused `LEGACY_BACKEND` build definitions

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test ci please

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test

File: tests/arm/Tests.lst [hunk 1]
```diff
@@ -83145,7 +83145,7 @@ RelativePath=JIT\Regression\CLR-x86-JIT\V1-M09\b15783\b15783\b15783.cmd
 WorkingDir=JIT\Regression\CLR-x86-JIT\V1-M09\b15783\b15783
 Expected=0
 MaxAllowedDurationSeconds=600
-Categories=EXPECTED_FAIL;13604;Pri1
```
Comment (sandreenko) [inline tests/arm/Tests.lst]: How is it connected with legacy backend?

Comment (BruceForstall) [reply tests/arm/Tests.lst]: ah, I noticed that issue was closed and it seemed the tests should pass now. What we should do is change it to `EXPECTED_PASS` and remove the issue. I'll just leave it as is for now.

**Pull request merged**